src/arrayMethodJoin.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        result += this[i];
+      }
+
+      if (i !== this.length - 1) {
+        if (separator === undefined) {
+          result += ',';
```

Comment:
You can remove this check if you set ',' as the default value for separator

Reply:
Thanks! Sometimes I make a much more difficult solution, that I need :)